client/landing/stepper/declarative-flow/internals/steps-repository/pattern-assembler/hooks/use-global-styles-upgrade-modal.ts

```diff
@@ -0,0 +1,57 @@
+import { useState } from 'react';
+import { urlToSlug } from 'calypso/lib/url';
+import { useSite } from '../../../../../hooks/use-site';
+import { useSiteSlugParam } from '../../../../../hooks/use-site-slug-param';
+
+interface Props {
+	recordTracksEvent: ( eventName: string, eventProps?: { [ key: string ]: unknown } ) => void;
+	onSubmit: () => void;
+}
+
+const useGlobalStylesUpgradeModal = ( { recordTracksEvent, onSubmit }: Props ) => {
+	const [ isOpen, setIsOpen ] = useState( false );
+	const site = useSite();
+	const siteSlug = useSiteSlugParam();
+	const siteUrl = siteSlug || urlToSlug( site?.URL || '' ) || '';
+
+	const openModal = () => {
+		recordTracksEvent( 'calypso_signup_pattern_assembler_global_styles_gating_modal_show' );
+		setIsOpen( true );
+	};
+
+	const closeModal = () => {
+		recordTracksEvent(
+			'calypso_signup_pattern_assembler_global_styles_gating_modal_close_button_click'
+		);
+		setIsOpen( false );
+	};
+
+	const checkout = () => {
+		recordTracksEvent(
+			'calypso_signup_pattern_assembler_global_styles_gating_modal_checkout_button_click'
+		);
+
+		// When the user is done with checkout, send them back to the current url
+		const destUrl = new URL( window.location.href );
+		const redirectUrl = destUrl.toString().replace( window.location.origin, '' );
+		const params = new URLSearchParams( {
+			redirect_to: redirectUrl,
```

Comment:
I noticed that closing the checkout page will redirect users to the Plans page:
Uploading Screen Recording on 2023-03-06 at 12_18_26.mp4…

I'm not totally sure if this is something we can address, but at least it's worth being aware of this, since it might impact the PA completion rate significantly.

Reply:
It's a long-term issue not just for the PA but also for upgrading to activate the design. That's why we brought up the checkout modal to the setup flow before but it doesn't work now 😔 Maybe we have to fix this issue once it has a significant impact to complete setup flow